File: Baekjoon/Sorting/[2822]점수 계산/solution.py
number = [0 for i in range(5)]
grade = [[] for i in range(8)]
res = 0

for i in range(8):
    point = int(input())
    grade[i] = [point, i + 1]

grade.sort(key=lambda x: x[0], reverse=True)

for i in range(5):
    res += grade[i][0]
    number[i] = grade[i][1]

print(res)

number.sort()

for n in number:
    print(n, end=" ")

File: Baekjoon/Sorting/[2822]점수 계산/solution.cpp
#include <algorithm>
#include <iostream>
#include <vector>

using namespace std;

#define endl "\n"

typedef pair<int, int> pii;

vector<pii> grade;
vector<int> question;
int res;

int main() {
  ios::sync_with_stdio(false);
  cin.tie(nullptr);

  grade = vector<pii>(8, {0, 0});
  res = 0;

  for (int i = 0; i < 8; i++) {
    cin >> grade[i].first;
    grade[i].second = i + 1;
  }

  sort(grade.begin(), grade.end(), greater<>());

  for (int i = 0; i < 5; i++) {
    res += grade[i].first;
  }

  cout << res << endl;

  for (int i = 0; i < 5; i++) {
    question.push_back(grade[i].second);
  }

  sort(question.begin(), question.end());

  for (int i = 0; i < 5; i++) {
    cout << question[i] << " ";
  }
  cout << endl;

  return 0;
}